### PR TITLE
Lot view with collection views 😍

### PR DESCRIPTION
**Pull request by ashfurrow**
This is initial work to achieve https://github.com/artsy/eigen/issues/1085 and unblock https://github.com/artsy/eigen/pull/1506 .

### CODE REVIEW

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift [hunk 1]
```diff
@@ -183,12 +210,44 @@ class LiveAuctionLotSetViewController: UIViewController {
...
         dismissViewControllerAnimated(true, completion: nil)
     }
+
+}
+
+private typealias HostScrollViewDelegate = LiveAuctionLotSetViewController
+extension HostScrollViewDelegate: UIScrollViewDelegate {
+
+    // The idea is to match the page view controller's scrollview's content offset to that of our collection view.
+    // The colleciton view data source mimics the page view controller's three-at-a-time display strategy.
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift]: colleciton

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift [hunk 2]
```diff
@@ -31,33 +43,48 @@ class LiveAuctionLotSetViewController: UIViewController {
...
         progressBar.alignBottomEdgeWithView(view, predicate: "-165")
 
+        // Final setup for our (now constructed) view hierarchy.
         setupWithInitialData()
     }
 
     override func viewWillAppear(animated: Bool) {
         super.viewWillAppear(animated)
+        // TODO: hand changing trait collections.
         setupToolbar()
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotSetViewController.swift]: dispatch once

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift [hunk 3]
```diff
@@ -141,10 +134,12 @@ class LiveAuctionLotViewController: UIViewController {
...
             // We need to align the bottom of the lot image to the lot metadata
             lotMetadataStack?.layoutIfNeeded()
-            let height = lotMetadataStack?.frame.height ?? 0
-            imageBottomConstraint.constant = height + 20
+            // TODO: needed?
+//            let height = lotMetadataStack?.frame.height ?? 0
+//            imageBottomConstraint.constant = height + 20
         }
 
+        // TODO: is this required? A closed sale would imply all lots are closed, and the currentLotView would be hidden in the above subscription ^
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift]: aye

File: Artsy/View_Controllers/Live_Auctions/LiveAuctionSalesPerson.swift [hunk 4]
```diff
@@ -73,10 +76,24 @@ private typealias PublicFunctions = LiveAuctionsSalesPerson
...
+    func lotViewModelRelativeToShowingIndex(offset: Int) -> LiveAuctionLotViewModelType {
+        precondition(abs(offset) < lotCount)
+
+        let currentlyShowingIndex = currentFocusedLotIndex.peek() ?? 0 // The coalesce is only to satisfy the compiler, should never happen since the currentFocusedLotIndex is created with an initial value.
+
+        // Apply the offset
         let newIndex = currentlyShowingIndex + offset
-        let loopingIndex = newIndex > 0 ? newIndex : lots.count + offset
+
+        // Guarnatee the offset is within the bounds of our array.
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/LiveAuctionSalesPerson.swift]: Guarnatee

File: Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionLotCollectionViewDataSource.swift [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+        self.salesPerson = salesPerson
+    }
+
+}
+
+private typealias CollectionViewDataSource = LiveAuctionLotCollectionViewDataSource
+extension CollectionViewDataSource: UICollectionViewDataSource {
+
+    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        // We always have three, for: previous, current, and next. We rely on the UIPageViewCotrollerDelegate callbacks
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionLotCollectionViewDataSource.swift]: UIPageViewCotrollerDelegate

colleciton

File: Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionLotImagesPreviewView.swift [hunk 6]
```diff
@@ -1,137 +0,0 @@
...
-    let progress = Observable<CGFloat>()
-
-    func scrollViewDidScroll(scrollView: UIScrollView) {
-        let point = scrollView.contentOffset
-
-        // Creates a value from -1 to 0 to 1
-        let index = (point.x - scrollView.frame.width) / scrollView.frame.width * -1;
-        progress.update(index)
-    }
-}
```
Comment (orta) [inline Artsy/View_Controllers/Live_Auctions/Views/LiveAuctionLotImagesPreviewView.swift]: byeeee

**Comment by orta** [comment]
Cool other than all those typoes

**Comment by ArtsyMobile** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-kind="Warning">
        
            2 Warnings
        
      </th>
     </tr>
  </thead>
  <tbody>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
        <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
    
    
  </tbody>
</table>
  

  



Time | Class | Function |
| --- | ----- | ----- |243.4ms | [AuctionViewController.swift](/artsy/eigen/tree/ashfurrow-lot-view/Artsy/View_Controllers/Auction/AuctionViewController.swift#L176) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
195.9ms | [SwiftyJSON.swift](/artsy/eigen/tree/ashfurrow-lot-view/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
171.9ms | [WebSocket.swift](/artsy/eigen/tree/ashfurrow-lot-view/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)




<p align="right" data-meta="generated_by_danger" data-base-commit="" data-head-commit="" >
  Generated by :no_entry_sign: <a href="https://github.com/danger/danger/">danger</a>
</p>

**Comment by orta** [comment]
👍

**Pull request merged**